src/components/side_panel/components/checkbox/checkbox.xml

```diff
@@ -7,6 +7,7 @@
         t-att-checked="props.value"
         t-on-change="onChange"
         class="align-middle"
+        t-att-disabled="props.disabled"
```

Comment:
label should also be grayed out when the checkbox is disabled, otherwise it's not really visible the checkbox is disabled (at least for my colorblind eyes). See [here](https://github.com/odoo/o-spreadsheet/pull/3536/files#diff-93c872bfb48cc2863dbb3f42e071e10f05eae47c29a78e769653da51874a4097)